tools/distpartitioning/data_shuffle.py

```diff
@@ -201,119 +343,84 @@ def exchange_features(rank, world_size, feature_tids, ntype_gnid_map, id_lookup,
+       Distributed lookup service used to map global-nids to respective 
+       partition-ids and shuffle-global-nids
     feat_type : string
-        this is used to distinguish which features are being exchanged. Please note that
-        for nodes ownership is clearly defined and for edges it is always assumed that
-        destination end point of the edge defines the ownership of that particular 
-        edge
+        this is used to distinguish which features are being exchanged. Please 
+        note that for nodes ownership is clearly defined and for edges it is 
+        always assumed that destination end point of the edge defines the 
+        ownership of that particular edge
+    data: dicitonary
+        dictionry in which node or edge features are stored and this information
+        is read from the appropriate node features file which belongs to the 
+        current process
 
     Returns:
     --------
     dictionary :
-        node features are returned as a dictionary where keys are node type names and node feature names
-        and values are tensors
-    dictionary :
-        a dictionary of global_nids for the nodes whose node features are received during the data shuffle
-        process
+        a dictionary is returned where keys are type names and 
+        feature data are the values
+    list :
+        a dictionary of global_ids either nodes or edges whose features are 
+        received during the data shuffle process
     """
     start = timer()
     own_features = {}
-    own_global_nids = {}
-    #To iterate over the node_types and associated node_features
-    for type_name, type_info in feature_tids.items():
-
-        #To iterate over the node_features, of a given node_type
-        #type_info is a list of 3 elements
-        #[feature-name, starting-idx, ending-idx]
-        #feature-name is the name given to the feature-data, read from the input metadata file
-        #[starting-idx, ending-idx) specifies the range of indexes associated with the features read from
-        #the associated input file. Note that the rows of features read from the input file should be same
-        #as specified with this range. So no. of rows = ending-idx - starting-idx.
-        for feat_info in type_info:
+    own_global_ids = {}
+
+    # To iterate over the node_types and associated node_features
+    for feat_key, type_info in feature_tids.items():
+
+        # To iterate over the feature data, of a given (node or edge )type
+        # type_info is a list of 3 elements (as shown below):
+        #   [feature-name, starting-idx, ending-idx]
+        #       feature-name is the name given to the feature-data, 
+        #       read from the input metadata file
+        #       [starting-idx, ending-idx) specifies the range of indexes 
+        #        associated with the features data 
+        # Determine the owner process for these features.
+        tokens = feat_key.split("/")
```

Comment:
it's `node_type/feat_name/local_part_id`? could you add a comment to showcase how `feat_key` is composed?

Reply:
Added a very detailed comment describing the format and meaning of local_part_id.